.github/actions/setup/action.yml

```diff
@@ -0,0 +1,35 @@
+name: "Setup & build"
+description: "Setup node, install dependencies & build"
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-node@v2
+      with:
+        node-version-file: '.nvmrc'
+    - uses: actions/cache@v2
```

Comment:
This action replaces the whole artifact download/upload and tar compress/extract logic I had in the last PR.

The cache downloads the build if it's available. If it's not available, it builds. The cache action also includes a post-job action which uploads the build to the cache if it wasn't downloaded.